torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -52,10 +73,16 @@ auto AccumulateGrad::apply(variable_list&& grads) -> variable_list {
       // If the function has post hooks (for example, a DDP allreduce hook),
       // call_function in Engine.cpp will temporarily bump the refcount by one, hence the
       // addition of !post_hooks().empty().
-      variable.grad() = new_grad.detach();
+      var.grad() = new_grad.detach();
     } else {
-      variable.grad() = new_grad.clone();
+      var.grad() = new_grad.clone();
     }
+    variable_grad = WeakVariable(var.grad()); // We need to update our reference
```

Comment:
Importantly, there is only ever one grad accumulator for a variable.

Reply:
Agreed, actually I didn't quite understand how the rest of the comment applies to this line.  I now think the rest is for the next line, so I'll remove it.